### PR TITLE
Create tests for Shareable

### PR DESCRIPTION
**Pull request by adklempner**

### CODE REVIEW

File: contracts/test-helpers/ShareableMock.sol [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+  }
+
+  function increaseCount(bytes32 action) onlymanyowners(action) {
+    count = count + 1;
+  }
+
+  function isOwnerConst(address _addr) constant returns (bool) {
+    return isOwner(_addr);
+  }
```
Comment (maraoz) [inline contracts/test-helpers/ShareableMock.sol]: Why did you need to add this function?

File: test/Shareable.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+  });
+
+  it('should only perform multisig function with enough sigs', async function() {
+    let requiredSigs = 3;
+    let owners = accounts.slice(1,4);
+    let shareable = await ShareableMock.new(owners, requiredSigs);
+    let hash = 1234;
+
+    let initCount = await shareable.count();
+    initCount = initCount.toString();
```
Comment (maraoz) [inline test/Shareable.js]: Using a string here seems confusing. What's the reason?

File: test/Shareable.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+  });
+
+  it('should require approval from different owners', async function() {
+    let requiredSigs = 2;
+    let owners = accounts.slice(1,4);
+    let shareable = await ShareableMock.new(owners, requiredSigs);
+    let hash = 1234;
+
+    let initCount = await shareable.count();
+    initCount = initCount.toString();
```
Comment (maraoz) [inline test/Shareable.js]: same here

File: contracts/test-helpers/ShareableMock.sol (hunk 1 above)
Comment (adklempner) [inline contracts/test-helpers/ShareableMock.sol]: Calling isOwner() in web3js does not return the function's return value because it's not const. Instead, it returns the transaction hash. Adding the isOwnerConst() function to the mock contract allows me to get around that.

File: test/Shareable.js (hunk 2 above)
Comment (adklempner) [inline test/Shareable.js]: I double checked and it doesn't make a difference. all the toString()'s in this test method can be removed.

File: test/Shareable.js (hunk 3 above)
Comment (adklempner) [inline test/Shareable.js]: Same as above

File: contracts/test-helpers/ShareableMock.sol (hunk 1 above)
Comment (maraoz) [inline contracts/test-helpers/ShareableMock.sol]: What I mean is, make `isOwner` constant, remove `isOwnerConst` from `ShareableMock`, and use `isOwner` directly in tests. In the future, feel free to change the base contracts when you're writing tests if you find this problems.
We want the code to be as simple/correct as possible.

**Comment by maraoz** [comment]
Added a comment, and please re-check tests that seem to be failing

**Comment by adklempner** [comment]
Looks like some of those toString() calls are necessary. Comparing the objects is what's causing the tests to fail: `AssertionError: expected { Object (s, e, ...) } to equal { Object (s, e, ...) }`

**Comment by maraoz** [comment]
LGTM
